python/dgl/utils.py

```diff
@@ -126,9 +127,61 @@ def __getstate__(self):
     def __setstate__(self, state):
         self._initialize_data(state)
 
+    def get_items(self, index):
+        tensor = self.tousertensor()
+        if isinstance(index, Index):
+            if isinstance(index._pydata, slice):
+                # XXX: assuming tensor supports index using slices
+                #      cannot call tousertensor because for slice type it would
+                #      further call tonumpy, which overwrites _pydata and causes
+                #      future bugs...
```

Comment:
What does this mean?

Reply:
Originally when I implemented get_items and set_items, I write 
```
if isintance(index, Index):
    index = index.tousertensor()
```
Recently I found this is buggy, because, if index._pydata is a slice, index.tousertensor will can tonumpy, which overwrites _pydata. As a result, index._pydata becomes a numpy.ndarray... And this can potentially cause bugs